plugins/out_s3/s3.c

```diff
@@ -986,11 +987,19 @@ static int upload_data(struct flb_s3 *ctx, struct s3_file *chunk,
     int size_check = FLB_FALSE;
     int part_num_check = FLB_FALSE;
     int timeout_check = FLB_FALSE;
-    time_t create_time;
     int ret;
     void *payload_buf = NULL;
     size_t payload_size = 0;
     size_t preCompress_size = 0;
+    time_t file_first_log_time = time(NULL);
+
+    /*
+     * When chunk does not exist, file_first_log_time will be 0.
```

Comment:
its no longer 0

Reply:
it'd be ideal if you stated why teh unit test will crash: because the chunk will be null in it